openmetadata-ui/src/main/resources/ui/src/components/AddService/Steps/SelectServiceType.tsx

```diff
@@ -96,32 +98,42 @@ const SelectServiceType = ({
           <div
             className="tw-grid tw-grid-cols-6 tw-grid-flow-row tw-gap-4 tw-mt-4"
             data-testid="select-service">
-            {selectedConnectors.map((type) => (
-              <div
-                className={classNames(
-                  'tw-flex tw-flex-col tw-items-center tw-relative tw-p-2 tw-w-24 tw-cursor-pointer tw-border tw-rounded-md',
-                  {
-                    'tw-border-primary': type === selectServiceType,
-                  }
-                )}
-                data-testid={type}
-                key={type}
-                onClick={() => handleServiceTypeClick(type)}>
-                <div className="tw-mb-2.5">
-                  <div data-testid="service-icon">
-                    {getServiceLogo(type || '', 'tw-h-9')}
-                  </div>
-                  <div className="tw-absolute tw-top-0 tw-right-1.5">
-                    {type === selectServiceType && (
-                      <SVGIcons alt="checkbox" icon={Icons.CHECKBOX_PRIMARY} />
-                    )}
+            {selectedConnectors
+              .filter(
+                (connectorType) =>
+                  !excludedService.includes(
+                    connectorType as MlModelServiceType | MetadataServiceType
+                  )
+              )
+              .map((type) => (
```

Comment:
Please extract this logic to above using `useMemo` to optimise it with re-renders

Reply:
Done